kpops/components/base_components/models/from_section.py

```diff
@@ -38,19 +42,39 @@ class Config(DescConfig):
 
     @root_validator
     def extra_topic_role(cls, values: dict) -> dict:
-        """Ensure that cls.role is used correctly"""
-        is_extra_topic = values["type"] in (
-            InputTopicTypes.EXTRA,
-            InputTopicTypes.EXTRA_PATTERN,
-        )
-        if is_extra_topic and not values.get("role"):
-            raise ValueError(
-                "If you define an extra input component, extra input topic, or extra input pattern, you have to define a role."
-            )
-        if not is_extra_topic and values.get("role"):
-            raise ValueError(
-                "If you do not define an input component, input topic, or input pattern, the role is unnecessary."
-            )
+        """Ensure that cls.role is used correctly, assign type if needed"""
+        type_ = values.get("type", None)
+        role = values.get("role", None) is not None
```

Comment:
why are you using `.get(...)`? they are always defined

Reply:
I think I got confused being defined as None with being undefined. Will fix